src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -766,3 +781,38 @@ method onFirstUnseenMessageLoaded*(self: Module, messageId: string) =
   self.firstUnseenMessageState.initialized = true
   self.firstUnseenMessageState.fetching = false
   self.reevaluateViewLoadingState()
+
+method forceLinkPreviewsLocalData*(self: Module, messageId: string) =
+  let item = self.view.model().getItemWithMessageId(messageId)
+  debug "forceLinkPreviewsLocalData", messageId, itemFound = $(item != nil)
+  if item == nil:
+    return
+  if not item.linkPreviewModel.updateForcedLocalDataTimestamp():
+    return
+  self.updateLinkPreviewsContacts(item, true)
+  self.updateLinkPreviewsCommunities(item, true)
+
+proc updateLinkPreviewsContacts(self: Module, item: Item, requestFromMailserver: bool) =
+  for contactId in item.linkPreviewModel.getContactIds().items:
+    let contact = self.controller.getContactDetails(contactId)
+    if contact.dto.displayName != "":
+      item.linkPreviewModel.setContactInfo(contact)
+      continue
+    if requestFromMailserver:
+      debug "updateLinkPreviewsContacts: contact not found, requesting from mailserver", contactId
+      item.linkPreviewModel.onContactDataRequested(contactId)
+      self.controller.requestContactInfo(contactId)
+  
+proc updateLinkPreviewsCommunities(self: Module, item: Item, requestFromMailserver: bool) =
+  for communityId in item.linkPreviewModel.getCommunityIds().items:
+    
+    if (let community = self.controller.getCommunityById(communityId); community).id != "":
+      item.linkPreviewModel.setCommunityInfo(community)
```

Comment:
Should we return/continue after this?

Reply:
Not really. We want to force the data to be fetched and updated when `requestFromMailserver == true`.

It's very possible that the community is not joined/spectated, so we don't get updates about it. But when we see that message with a preview in the chat, we for the data to be fetched from mailserver even if we have it locally.